src/Tizen.NUI/src/internal/Xaml/VisualState/VisualStateGroupList.cs

```diff
@@ -0,0 +1,130 @@
+﻿/*
+ * Copyright(c) 2021 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Tizen.NUI;
+using Tizen.NUI.Binding;
+
+namespace Tizen.NUI.Xaml
+{
+    internal class VisualStateGroupList : IList<VisualStateGroup>
+    {
+        readonly IList<VisualStateGroup> _internalList;
+
+        void Validate(IList<VisualStateGroup> groups)
+        {
+            // If we have 1 group, no need to worry about duplicate group names
+            if (groups.Count > 1)
+            {
+                if (groups.GroupBy(vsg => vsg.Name).Any(g => g.Count() > 1))
+                {
+                    throw new InvalidOperationException("VisualStateGroup Names must be unique");
+                }
+            }
+
+            // State names must be unique within this group list, so pull in all 
+            // the states in all the groups, group them by name, and see if we have
+            // and duplicates
+            if (groups.SelectMany(group => group.States).GroupBy(state => state.Name).Any(g => g.Count() > 1))
+            {
+                throw new InvalidOperationException("VisualState Names must be unique");
+            }
+        }
+
+        public VisualStateGroupList()
+        {
+            _internalList = new WatchAddList<VisualStateGroup>(Validate);
```

Comment:
please do not use `_` 
could you check this NUI coding convention?
https://code.sec.samsung.net/confluence/pages/viewpage.action?pageId=191373542